File: UESTC/2326.cpp
#include <bits/stdc++.h>
#define MAXX 205
#define MAXY 20105
#define MAXN 10005
using namespace std;

int n,k,b,cnt[MAXX][MAXY],ans[MAXN];

int main()
{
	scanf("%d",&n);
	for (int i=1;i<=n;i++)
	{
		scanf("%d %d",&k,&b);
		for (int j=0;j<=200;j++) ++cnt[j][k*j+b];
	}
	for (int i=0;i<=200;i++)
		for (int j=0;j<=20100;j++)
			++ans[cnt[i][j]];
	for (int i=2;i<=n;i++) printf("%d%c",ans[i],i==n?'\n':' ');
	return 0;
}

File: UESTC/2326.java
import java.util.*;

public class Main {
	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		int n = in.nextInt();
		int[] ans = new int[n + 1];
		int[][] cnt = new int[101][10105];
		for (int i = 1; i <= n; i++) {
			int k = in.nextInt();
			int b = in.nextInt();
			for (int j = 0; j <= 100; j++)
				++cnt[j][k * j + b];
		}
		for (int i = 0; i <= 100; i++)
			for (int j = 0; j <= 10100; j++)
				++ans[cnt[i][j]];
		for (int i = 2; i <= n; i++) {
			if (i != 2) System.out.print(' ');
			System.out.print(ans[i]);
		}
	}
}